lib/msal-common/src/client/SilentFlowClient.ts

```diff
@@ -60,14 +59,23 @@ export class SilentFlowClient extends BaseClient {
         const environment = request.authority || this.authority.getPreferredCache();
         const cacheRecord = this.cacheManager.readCacheRecord(request.account, this.config.authOptions.clientId, requestScopes, environment);
 
-        if (this.isRefreshRequired(request, cacheRecord.accessToken)) {
+        if (request.forceRefresh || 
+            request.claims || 
+            !cacheRecord.accessToken || 
+            TimeUtils.isTokenExpired(cacheRecord.accessToken.expiresOn, this.config.systemOptions.tokenRenewalOffsetSeconds)) {
+            // Must refresh due to request parameters, or expired or non-existent access_token
             throw ClientAuthError.createRefreshRequiredError();
-        } else {
-            if (this.config.serverTelemetryManager) {
-                this.config.serverTelemetryManager.incrementCacheHits();
-            }
-            return await this.generateResultFromCacheRecord(cacheRecord, request.resourceRequestMethod, request.resourceRequestUri);
+        } else if (cacheRecord.accessToken.refreshOn && TimeUtils.isTokenExpired(cacheRecord.accessToken.refreshOn, 0)) {
+            // Refresh on time has passed, start token renewal process and return current access token
+            const refreshTokenClient = new RefreshTokenClient(this.config);
+            refreshTokenClient.acquireTokenByRefreshToken(request);
```

Comment:
Make sure all errors from this call are caught.

Reply:
done, caught by calling function